### PR TITLE
Allow additional overrides

### PR DESCRIPTION
**Pull request by erik-hooper**
Allow BackchannelHandler and RefreshTokenMessageHandler overrides.
Allow additional parameters to RefreshTokenAsync.

### CODE REVIEW

File: src/Auth0.OidcClient.Shared/Auth0Client.cs [hunk 1]
```diff
@@ -194,11 +204,23 @@ public Task<LoginResult> ProcessResponseAsync(string data, AuthorizeState state)
...
+        }
+
+        /// <summary>
+        /// Generates a new set of tokens based on a refresh token. 
+        /// </summary>
+        /// <param name="refreshToken">The refresh token which was issued during the authorization flow, or subsequent
+        /// calls to <see cref="OidcClient.RefreshTokenAsync"/>.</param>
+        /// <param name="extraParameters">Additional parameters to send to the refresh endpoint</param>
+        /// <returns></returns>
+        public Task<RefreshTokenResult> RefreshTokenAsync(string refreshToken, object extraParameters)
```
Comment (damieng) [inline src/Auth0.OidcClient.Shared/Auth0Client.cs]: It looks like RefreshTokenAsync has got into this PR by mistake?  I thought it was just supposed to be the two options for the handlers.

File: src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs [hunk 2]
```diff
@@ -49,6 +50,30 @@ public class Auth0ClientOptions
...
+        /// <summary>
+        /// Allow overriding the RetryMessageHandler
+        /// <example>
+        /// var handler = new HttpClientHandler();
+        /// var options = new Auth0ClientOptions
+        /// {
+        ///    RefreshTokenMessageHandler = handler
+        /// };
+        /// </example>
+        /// </summary>
```
Comment (damieng) [inline src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs]: Summary tag should close before the example tag.

File: src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs [hunk 3]
```diff
@@ -49,6 +50,30 @@ public class Auth0ClientOptions
...
+        /// <summary>
+        /// Allow overriding the BackchannelHandler
+        /// <example>
+        /// var handler = new HttpClientHandler();
+        /// var options = new Auth0ClientOptions
+        /// {
+        ///    BackchannelHandler = handler
+        /// };
+        /// </example>
+        /// </summary>
```
Comment (damieng) [inline src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs]: Summary tag should close before the example tag.

File: src/Auth0.OidcClient.Shared/Auth0Client.cs (hunk 1 above)
Comment (erik-hooper) [reply src/Auth0.OidcClient.Shared/Auth0Client.cs]: No I meant to include this as well. For refresh we needed to provide some additional parameters for some auth providers.(Microsoft as I remember)

File: src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs (hunk 2 above)
Comment (erik-hooper) [reply src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs]: I've fixed this with another push sorry about that

File: src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs (hunk 3 above)
Comment (erik-hooper) [reply src/Auth0.OidcClient.Shared/Auth0ClientOptions.cs]: I've fixed this with another push sorry about that

File: src/Auth0.OidcClient.Shared/Auth0Client.cs (hunk 1 above)
Comment (damieng) [reply src/Auth0.OidcClient.Shared/Auth0Client.cs]: Adding a method to an interface is technically a breaking change.  I'm tempted to remove the interface entirely - it doesn't serve any useful purpose.

Comment (erik-hooper) [reply src/Auth0.OidcClient.Shared/Auth0Client.cs]: You are right it is a breaking change. Hopefully only for mocks or unit tests though. It is nice to have that abstraction though. Maybe remove the interface change and just have the implementation have that change?

**Comment by erik-hooper** [comment]
Thank you!

**Pull request merged**